NGitLab/IGroupsClient.cs

```diff
@@ -29,6 +29,8 @@ public interface IGroupsClient
 
         IEnumerable<Project> SearchProjects(int groupId, string search);
 
+        IEnumerable<Project> SearchProjects(string groupId, string search, GroupQueryScope scope);
```

Comment:
When we add new overloads in the interface, we try to use a class. This way, we don't need to re-update it if we want to add a new parameter.
Also, as this is a new class, you should use properties (`{get;set}`) instead of fields.

```suggestion
        IEnumerable<Project> SearchProjects(SearchProjectQuery query);
```

Reply:
According to the documentation examples, changing the `scope` also change the result type. For instance, `scope=merge_requests` returns a list of merge request. If so, we cannot allow any scope and return a list of projects.

Reply:
Very valid comment. Shall we reduce now the GroupQueryScope options again to allow only the proper return types?

Reply:
If I'm not mistaken, the only valid option (to return a collection of projects) is `project`, but there is already this overload that does exactly that `IEnumerable<Project> SearchProjects(int groupId, string search);`